service/stock/src/test/java/nl/tudelft/wdm/group1/stock/event/StockConsumerMultiThreadTest.java

```diff
@@ -0,0 +1,83 @@
+package nl.tudelft.wdm.group1.stock.event;
+
+import com.anarsoft.vmlens.concurrent.junit.ConcurrentTestRunner;
+import nl.tudelft.wdm.group1.common.*;
+import nl.tudelft.wdm.group1.stock.StockItemRepository;
+import nl.tudelft.wdm.group1.stock.events.Consumer;
+import nl.tudelft.wdm.group1.stock.events.Producer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@RunWith(ConcurrentTestRunner.class) // default # threads = 4
+public class StockConsumerMultiThreadTest {
+    private Producer stockProducer;
+    private Consumer stockConsumer;
+    private StockItemRepository stockItemRepository;
+    private StockItem stockItemMilk;
+    private StockItem stockItemJuice;
+    private StockItem stockItemCoke;
+
+    @Before
+    public void setUp() {
+        stockProducer = mock(Producer.class);
+        stockItemRepository = mock(StockItemRepository.class);
+        stockConsumer = new Consumer(stockItemRepository, stockProducer);
+
+        stockItemMilk = new StockItem(5, "Milk", 1);
+        stockItemJuice = new StockItem(4, "Juice", 2);
+        stockItemCoke = new StockItem(3, "Coke", 3);
+    }
+
+
+    @Test
+    public void testHandleOrderCheckedOutWithSufficientStockForMultipleUsers()
```

Comment:
What does this test exactly test? And how does the concurrency work? When I remove the locks from the consumer the test still passes.